pathos/tests/test_mp.py

```diff
@@ -27,6 +28,15 @@ def test_mp():
     result = result_queue.get()
     assert result == _result
 
+    # test ProcessPool keyword argument propagation
+    pool.clear()
+    pool = ProcessPool(nodes=4, initializer=lambda: time.sleep(0.6))
```

Comment:
```suggestion
    pool = ProcessPool(nodes=4, initializer=lambda: time.sleep(0.6))
    assert pool._pool.initializer, 'Subsequent pool with different kwds should propagate'
```
right? if not propagated, default initializer should be falsely?

wondering why this test actually takes 0.6+ seconds, doesn't that mean propagation is working? :thinking:

Reply:
Default initializer is None, I believe.  You can check the defaults with:
```
>>> from pathos.pools import _ProcessPool as Pool
>>> p = Pool()
>>> p._initializer

```
and so on.

Reply:
is there a state bleed somewhere? because by your theory, my tests should be failing. as the test does exactly what you described: first a plain pool, then a pool with initializer. by your theory, that initializer should be ignored and old pool reused. but by the test, the map now takes much longer. so apparently the initializer is propagated?

Reply:
I don't think you've made all the necessary edits. See comments above.

Reply:
sure, I'll still make the changes, but I still would like to understand why the test is showing the expected results. L13 populates the state, and L33 calls ProcessPool() again but with sleep initializer. And then `time.monotonic` shows that the call to map now indeed takes longer due to the sleep. So why does it currently work, if you [say](https://github.com/uqfoundation/pathos/pull/252#pullrequestreview-1223385469) it should not work?

Reply:
It was probably clearing the cached pool because `_serve` was being called with empty `kwds` in the `map` call (and the proper state handling wasn't done correctly).